Core/Core/Src/main.c

```diff
@@ -0,0 +1,221 @@
+/* Includes ------------------------------------------------------------------*/
+#include "main.h"
+#include "spi.h"
+#include "tim.h"
+#include "usart.h"
+#include "gpio.h"
+#include "stm32f0xx_hal_tim.h"
+
+/* Private includes ----------------------------------------------------------*/
+/* USER CODE BEGIN Includes */
+
+/* USER CODE END Includes */
+
+/* Private typedef -----------------------------------------------------------*/
+/* USER CODE BEGIN PTD */
+
+/* USER CODE END PTD */
+
+/* Private define ------------------------------------------------------------*/
+/* USER CODE BEGIN PD */
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+
+/* USER CODE BEGIN PV */
+
+
+/* USER CODE END PV */
+
+/* Private function prototypes -----------------------------------------------*/
+void SystemClock_Config(void);
+/* USER CODE BEGIN PFP */
+
+/* USER CODE END PFP */
+
+/* Private user code ---------------------------------------------------------*/
+/* USER CODE BEGIN 0 */
+
+/* USER CODE END 0 */
+
+/**
+  * @brief  The application entry point.
+  * @retval int
+  */
+int main(void)
+{
+  /* USER CODE BEGIN 1 */
+
+	static const uint16_t MAX_ADC_VAL = 1023;
+	static const uint16_t MAX_TIM_VAL = 65535;
+	static const uint16_t TIMEOUT = 100; //arbitrary ms value because not sure what to put
+	const uint16_t MIN_DUTY_CYCLE = 0.05;
+	static const uint16_t MIN_COMPARE_VALUE = MAX_TIM_VAL*MIN_DUTY_CYCLE;
+
+	static uint16_t VAL = 0x0;
+	static uint16_t COUNT_VAL = 0x0;
+	static uint16_t COMPARE_VAL = 0;
+	static const RX_BUFFER_SIZE = 3;   //allocation for receieing data buffer
+	static const uint8_t SPI_P_TX[3] ={0x01,0x80,0x00};  //allocation for transmit data buffer in CH0
+	uint8_t SPI_P_RX[3];  //allocation for receiving data buffer
+
+  /* USER CODE END 1 */
+
+  /* MCU Configuration--------------------------------------------------------*/
+
+  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+  HAL_Init();
+
+  /* USER CODE BEGIN Init */
+
+  /* USER CODE END Init */
+
+  /* Configure the system clock */
+  SystemClock_Config();
+
+  /* USER CODE BEGIN SysInit */
+
+  /* USER CODE END SysInit */
+
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+  MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM2_Init();
+  /* USER CODE BEGIN 2 */
+
+  /* USER CODE END 2 */
+
+  /* Infinite loop */
+  /* USER CODE BEGIN WHILE */
+
+
+  while (1)
+  {
+
+	 HAL_TIM_PWM_Start(&htim2, TIM_CHANNEL_2); //starting the PWM modulation
+
+	 HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET); //pulling line high in case it started low
+	 HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET); //pulling line low to begin communications
+
+	 HAL_SPI_TransmitReceive(&hspi1, SPI_P_TX, SPI_P_RX,sizeof(RX_BUFFER_SIZE), TIMEOUT);

```

Comment:
So here, instead of setting the line high then low right after, outside the loop you would set the line high to start to make sure it is high. THEN when you want to do a spi transmission/receive you would first put the line low, call HAL_SPI_TransmitReceive, then put the line high right after.